### PR TITLE
Add the ability to import checklists from Trello

### PR DESCRIPTION
**Pull request by leosunmo**
#### Summary
Adds the ability to import checklists to Focalboard when importing from Trello. It's pretty simple code, but it's my first attempt at TypeScript, so appreciate any feedback. I'm not 100% sure of my `undefined` logic etc. But it does work :shrug: 

This is kind of related to https://github.com/mattermost/focalboard/issues/53 but there's no specific issue relating to this. I wrote it for my own needs and figured I should push it upstream.

I started looking at importing images as well, but they require auth now if the Trello board is private. That is a different kettle of fish, so decided to leave that out of this PR.

### CODE REVIEW

File: import/trello/importTrello.ts [hunk 1]
```diff
@@ -135,6 +136,29 @@ function convert(input: Trello): Block[] {
...
+
+        // Add Checklists
+        if (card.idChecklists && card.idChecklists.length) {
+            card.idChecklists.forEach(checklistID => {
+                const lookup = input.checklists.find(e => e.id === checklistID)
+                if (lookup !== undefined) {
+                    lookup.checkItems.forEach(trelloCheckBox=> {
+                        const checkBlock = createCheckboxBlock()
+                        checkBlock.title = trelloCheckBox.name
+                        if (trelloCheckBox.state == 'complete') {
```
Comment (chenilim) [inline import/trello/importTrello.ts:148]: The eslint setup for the import scripts may be out of date from the webapp, but in general we should use ===, i.e. `if (trelloCheckBox.state === 'complete') {`.

File: import/trello/importTrello.ts [hunk 2]
```diff
@@ -135,6 +136,29 @@ function convert(input: Trello): Block[] {
 
             outCard.fields.contentOrder = [text.id]
         }
+
+        // Add Checklists
+        if (card.idChecklists && card.idChecklists.length) {
+            card.idChecklists.forEach(checklistID => {
+                const lookup = input.checklists.find(e => e.id === checklistID)
+                if (lookup !== undefined) {
```
Comment (chenilim) [inline import/trello/importTrello.ts:144]: In general, we can check if a variable is "tuthy" i.e. `if (lookup) {`, which will handle undefined, null, and other cases that can lead to a null reference exception.

File: import/trello/importTrello.ts [hunk 3]
```diff
@@ -135,6 +136,29 @@ function convert(input: Trello): Block[] {
 
             outCard.fields.contentOrder = [text.id]
         }
+
+        // Add Checklists
+        if (card.idChecklists && card.idChecklists.length) {
```
Comment (chenilim) [inline import/trello/importTrello.ts:141]: This is minor, and may be stylistic, but in general lengths should be checked as > 0, i.e. ` if (card.idChecklists && card.idChecklists.length > 0) {`.

File: import/trello/importTrello.ts [hunk 4]
```diff
@@ -9,7 +9,8 @@ import {IPropertyOption, IPropertyTemplate, createBoard} from '../../webapp/src/
 import {createBoardView} from '../../webapp/src/blocks/boardView'
 import {createCard} from '../../webapp/src/blocks/card'
 import {createTextBlock} from '../../webapp/src/blocks/textBlock'
-import {Trello} from './trello'
+import {createCheckboxBlock, CheckboxBlock} from '../../webapp/src/blocks/checkboxBlock'
```
Comment (chenilim) [inline import/trello/importTrello.ts:12]: ESlint called out that `CheckboxBlock` and `ChecklistElement` aren't used, and should not be imported.

**Review by chenilim — CHANGES_REQUESTED** [review]
Thanks @leosunmo for your contribution! This is a great addition to the import script! Logic looks good, just a few minor linter errors - the import scripts aren't part of the main CI / lint check.

File: import/trello/importTrello.ts (hunk 3 above)
Comment (leosunmo) [reply import/trello/importTrello.ts:141]: That was another thing I was wondering about. I originally used `> 0` (habit from writing Go) but kept seeing people doing what I ended up doing :shrug: so I wasn't sure if there was some hidden syntactic thing or just a style thing. I definitely prefer the clarity of `> 0`.

File: import/trello/importTrello.ts (hunk 4 above)
Comment (leosunmo) [reply import/trello/importTrello.ts:12]: Yeah my editor says the same, not sure why they're still there to be honest, must've had a language server error when I tidied and committed. Had to set up the TypeScript/JavaScript environment from scratch for this.

**Comment by leosunmo** [comment]
Take a look at my pushed changed. Pretty sure I addressed all your concerns. Really appreciate the language/style pointers, as I said, I've stayed away from JS/TS :p

**Review by chenilim — APPROVED** [review]
Thanks @leosunmo! LGTM 👍

**Comment by emilyacook** [comment]
Thanks for participating in Hacktoberfest! You can claim your sticker set here: https://get.printfection.com/hacktober21/4144583267 @leosunmo

**Pull request merged**